karabo_data/geometry2/__init__.py

```diff
@@ -949,23 +973,36 @@ def plot_data(self, modules_data, axis_units='px', frontview=True):
             raise ValueError("axis_units must be 'px' or 'm', not {!r}"
                              .format(axis_units))
 
-        fig = plt.figure(figsize=(10, 10))
-        ax = fig.add_subplot(1, 1, 1)
-        my_viridis = copy(viridis)
-        # Use a dark grey for missing data
-        my_viridis.set_bad('0.25', 1.0)
+        cbar_args = cbar_args or {}
 
         res, centre = self.position_modules(modules_data)
         min_y, min_x = -centre
         max_y, max_x = np.array(res.shape) - centre
 
-        extent = np.array((min_x - 0.5, max_x + 0.5, min_y - 0.5, max_y + 0.5))
+        _extent = np.array((min_x - 0.5, max_x + 0.5, min_y - 0.5, max_y + 0.5))
         cross_size = 20
         if axis_units == 'm':
-            extent *= self.geom.pixel_size
+            _extent *= self.geom.pixel_size
             cross_size *= self.geom.pixel_size
 
-        ax.imshow(res, origin='lower', cmap=my_viridis, extent=extent)
+        # Use a dark grey for missing data
+        _cmap = copy(viridis)
+        _cmap.set_bad('0.25', 1.0)
+        for key, default in (('extent', _extent),
+                             ('cmap', _cmap),
+                             ('origin', 'lower')):
+            # Set some default key word arguments for the plot
+            kwargs[key] = kwargs.get(key, default)
```

Comment:
I think you can use `kwargs.setdefault()` here ([docs](https://docs.python.org/3/library/stdtypes.html#dict.setdefault)).

I'd also get rid of the for loop - it takes more lines as a loop than without one.